filters/include/pcl/filters/impl/farthest_point_sampling.hpp

```diff
@@ -37,71 +35,92 @@
  *
  */
 
-#ifndef PCL_FILTERS_FARTHEST_POINT_SAMPLING_IMPL_H_
-#define PCL_FILTERS_FARTHEST_POINT_SAMPLING_IMPL_H_
+#pragma once
 
-#include <limits>
-#include <algorithm>
-#include <pcl/filters/farthest_point_sampling.h>
 #include <pcl/common/geometry.h>
+#include <pcl/filters/farthest_point_sampling.h>
 #include <pcl/point_cloud.h>
 #include <pcl/point_types.h>
-
+#include <algorithm>
+#include <limits>
+#include <random>
 
 template<typename PointT> void
 pcl::FarthestPointSampling<PointT>::applyFilter (PointCloud &output)
 {
-    int initial_size = input_->points.size();
-    int sample_size = sample_;
-    int max_index;
-    std::vector<int> selected_indices;
-    std::vector<float>distances_to_selected_points;
-    
-    //set random seed
-    std::srand (seed_);
+  std::vector<int> indices;
```

Comment:
We can reserve the size, since we know the size of the indices (filtered or max)